File: frontend/src/types/todo.ts
export type Todo = {
  id: number;
  title: string;
  done: boolean;
  createdAt: string;
  updatedAt: string;
  priority: number;
  dueAt?: string | null;
};

export type UpdatePayload = {
  title: string;
  done: boolean;
  priority?: number;
  due_at?: string | null;
};

export type SortKey = "priority" | "title" | "dueAt" | "updatedAt";

export type SortDirection = "asc" | "desc";

File: internal/domain/todo.go
package domain

import "time"

type Todo struct {
	ID        uint       `json:"id"`
	Title     string     `json:"title"`
	Done      bool       `json:"done"`
	CreatedAt time.Time  `json:"created_at"`
	UpdatedAt time.Time  `json:"updated_at"`
	Priority  uint8      `json:"priority" gorm:"not null;default:3"`
	DueAt     *time.Time `json:"due_at"`
}
